src/hidimstat/_utils/scenario.py

```diff
@@ -195,159 +146,156 @@ def multivariate_simulation(
     X_ = rng.standard_normal((n_samples,) + shape)
+    ## Create local correlation
     X = []
     for i in np.arange(n_samples):
         Xi = ndimage.gaussian_filter(X_[i], smooth_X)
         X.append(Xi.ravel())
-
     X = np.asarray(X)
-    X_ = X.reshape((n_samples,) + shape)
 
-    noise = sigma * rng.standard_normal(n_samples)
+    # Generate the support and the noise of the data for the prediction.
+    noise = sigma_noise * rng.standard_normal(n_samples)
+    # generate the support of the data
+    if len(shape) == 2:
+        w = _generate_2D_weight(shape, roi_size)
+    elif len(shape) == 3:
+        w = _generate_3D_weight(shape, roi_size)
+    else:
+        raise ValueError(f"Shape {shape} not supported, only 2D and 3D are supported")
+    beta = w.sum(-1).ravel()
+
     y = np.dot(X, beta) + noise
 
-    return X, y, beta, noise, X_, w
+    return X, y, beta, noise
 
 
-def multivariate_temporal_simulation(
-    n_samples=100,
-    n_features=500,
-    n_times=30,
+def multivariate_simulation(
+    n_samples,
+    n_features,
+    n_targets=None,
     support_size=10,
-    sigma=1.0,
-    rho_noise=0.0,
-    rho_data=0.0,
-    shuffle=True,
-    seed=0,
+    rho=0,
+    value=1.0,
+    signal_noise_ratio=10.0,
+    rho_serial=0.0,
+    shuffle=False,
+    continuous_support=False,
+    seed=None,
 ):
     """
-    Generate 1D temporal data with constant design matrix.
+    Generate a linear simulation with a Toeplitz covariance structure and optional temporal correlation.
 
     Parameters
     ----------
-    n_samples : int, default=100
+    n_samples : int
         Number of samples.
-    n_features : int, default=500
-        Number of features.
-    n_times : int, default=30
-        Number of time points.
+    n_features : int
+        Number of features/variables.
+    n_targets : int or None, default=None
+        Number of targets/time points. If None, generates univariate target data.
+        If integer > 0, generates multivariate target data.
     support_size : int, default=10
-        Size of the row support (number of non-zero coefficient rows).
-    sigma : float, default=1.0
-        Standard deviation of the additive white Gaussian noise.
-    rho_noise : float, default=0.0
-        Level of temporal autocorrelation in the noise. Must be between 0 and 1.
-    rho_data : float, default=0.0
-        Level of correlation between neighboring features. Must be between 0 and 1.
-    shuffle : bool, default=True
-        If True, randomly shuffle the features to break 1D data structure.
-    seed : int, default=0
+        Number of non-zero coefficients in beta.
+    rho : float, default=0
+        Feature correlation coefficient for Toeplitz covariance matrix.
+    value : float, default=1.0
+        Value assigned to non-zero coefficients in beta.
+    signal_noise_ratio : float, default=10.0
+        Signal-to-noise ratio. Controls noise scaling.
+    sigma_noise : float, default=1.0
```

Comment:
It is weird to specify both `sigma_noise` and `signal_noise ratio` ?

Reply:
I forget to remove `sigma_noise` from the docstring.